### ISSUE TITLE
publish to localhost, install from registry.mycompany.name -> invalid links

### ISSUE THREAD

**Issue by peritus** [npm/npm-registry-couchapp#28]
tl;dr:
- publish to localhost
- install from registry.mycompany.name
- GET /.../package-$version.tgz fails because of links point to localhost

---

Hey, here's what I experience — and i'm not sure whether this is a bug or a feature:

On my continous integration server (that happens to be the same machine as the one hosting CouchDB for our internal npm registry) publishes our internal packages like this:

```
NPM_CONFIG_REGISTRY=http://localhost:5984/registry/_design/app/_rewrite npm publish
```

Then, on my production machines, I install them like this:

```
NPM_CONFIG_REGISTRY=http://registry.mycompany.name:5984/registry/_design/app/_rewrite npm install mypkg
```

However, `registry/app.js` generates links at ``http://registry.mycompany.name:5984/registry/_design/app/_rewrite/mypkg` like this:

```
.. ..
dist: {
  shasum: "d9090586650b43865ae5d1bfade49d8d613b86e6"
  tarball: "http://localhost:5984/registry/_design/app/_rewrite/mypkg/-/mypkg-0.9.17pre89-5b31b8a.tgz"
}
.. ..
```

that, when `npm install` tries to resolve them, don't work (unless I manually `s/localhost/registry.mycompany.name/`  them via CouchDB's admin interface).

The code at https://github.com/isaacs/npmjs.org/blob/310d9afa176a076901bd2698250019c8668d7f33/registry/app.js#L365-370 gives me the impression there should be some rewriting going on at rendering time.

---

Workaround probably is to not publish to localhost but to the dns name that is also used later, but this might be part of a larger problem affecting more people, not sure.

**Comment by stepancar** [comment]
@peritus , hello! I have the same issue. "However, registry/app.js generates links at 
``` javascript
http://registry.mycompany.name:5984/registry/_design/app/_rewrite/mypkg
```
like this
``` javascript
dist: {
  shasum: "d9090586650b43865ae5d1bfade49d8d613b86e6"
  tarball: "http://localhost:5984/registry/_design/app/_rewrite/mypkg/-/mypkg-0.9.17pre89-5b31b8a.tgz"
}
```
what is simple solution of this problem?
@jimisaacs  why you not closed this issue?
Thank you!
